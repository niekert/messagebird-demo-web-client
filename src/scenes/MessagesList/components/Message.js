import React from 'react';
import { ContentCard } from 'style/Cards';
import { SentIcon, ReceivedIcon } from 'shared/components/icons';
import TimeAgo from 'shared/components/TimeAgo';
import { string, number, oneOf } from 'prop-types';
import styled from 'styled-components';
import { prop } from 'styled-tools';

const StyledCard = styled(ContentCard)`
  display: flex;
  align-items: center;
  height: 110px;
`;

const IconWrapper = styled.div`
  width: 24px;
  height: 24px;
  margin-right: ${prop('theme.spacing.1')};
`;

const Content = styled.div`
  padding: ${prop('theme.spacing.0')} 0;
  display: flex;
  flex: 1;
  flex-direction: column;
`;

const Body = styled.p`
  margin: 0;
`;

const Label = styled.span`
  font-size: 12px;
  font-weight: 600;
  margin-bottom: ${prop('theme.spacing.0')};
`;

const Time = styled(TimeAgo)`
  font-size: 10px;
  margin-left: ${prop('theme.spacing.1')};
  opacity: 0.8;
`;

function Message({ direction, recipient, body, originator, createdDatetime }) {
  const isSent = direction === 'mt';

  return (
    <StyledCard>
      <IconWrapper>{isSent ? <SentIcon /> : <ReceivedIcon />}</IconWrapper>
      <Content>
        <Label>{isSent ? `To ${recipient}` : `From ${originator}`}</Label>
        <Body>{body}</Body>
      </Content>
      <Time>{createdDatetime}</Time>
    </StyledCard>
  );
}

Message.propTypes = {
  direction: oneOf(['mt', 'mo']).isRequired,
  recipient: number.isRequired,
  body: string.isRequired,
  createdDatetime: string.isRequired,
  originator: string.isRequired,
};

export default Message;
