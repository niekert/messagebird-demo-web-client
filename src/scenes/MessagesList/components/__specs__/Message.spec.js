import React from 'react';
import { shallow } from 'enzyme';
import Message from '../Message';

describe('<Message />', () => {
  test('renders correctly', () => {
    const wrapper = shallow(
      <Message
        direction="mt"
        recipient={30201}
        body="test message"
        originator="01230123"
        createdDatetime="mock-date-time"
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
