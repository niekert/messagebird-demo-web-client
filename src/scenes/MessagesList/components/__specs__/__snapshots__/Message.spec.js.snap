// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Message /> renders correctly 1`] = `
<Message__StyledCard>
  <Message__IconWrapper>
    <SentIcon
      fill="currentColor"
    />
  </Message__IconWrapper>
  <Message__Content>
    <Message__Label>
      To 30201
    </Message__Label>
    <Message__Body>
      test message
    </Message__Body>
  </Message__Content>
  <Message__Time>
    mock-date-time
  </Message__Time>
</Message__StyledCard>
`;
